src/main/java/com/epam/aidial/cfg/web/controller/ApplicationTypeSchemaController.java

```diff
@@ -39,12 +39,19 @@ public ApplicationTypeSchemaController(ApplicationTypeSchemaFacade applicationTy
     }
 
     @GetMapping(produces = MediaType.APPLICATION_JSON_VALUE)
-    public Collection<ApplicationTypeSchemaDto> get(@RequestParam(name = "id", required = false) String id) {
+    public ResponseEntity<?> get(@RequestParam(name = "id", required = false) String id,
+                                 @RequestHeader(value = "If-None-Match", required = false) String previousHash) {
         if (StringUtils.isEmpty(id)) {
-            return applicationTypeSchemaFacade.getAll();
+            return ResponseEntity.ok(applicationTypeSchemaFacade.getAll());
         }
-        ApplicationTypeSchemaDto schemaDto = applicationTypeSchemaFacade.get(id);
-        return schemaDto != null ? List.of(schemaDto) : Collections.emptyList();
+        var schemaDto = applicationTypeSchemaFacade.getSchemaWithHash(id);
+        if (schemaDto == null) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).build();
```

Comment:
looks like breaking change too. add to PR description and discuss with FE team

Reply:
A description of the method's behavior has been added to the pull request (PR) description.